pkg/composableschemadsl/compiler/importer.go

```diff
@@ -84,6 +84,15 @@ func importFile(importContext importContext) (*CompiledSchema, error) {
 	)
 }
 
-func constructFilePath(segments []string) string {
-	return path.Join(segments...) + SchemaFileSuffix
+// Take a filepath and ensure that it's local to the current context.
+func validateFilepath(path string) error {
+	// NOTE: This is slightly overly restrictive; it should theoretically be possible
+	// to take a given filepath and figure out whether it's local to the context where
+	// the compiler is being invoked, rather than whether it's local to the source
+	// folder of the current context. The assumption is that that won't matter
+	// right now, and we can fix it if we need to.
+	if !filepath.IsLocal(path) {
```

Comment:
we should also add a check for `..` anywhere in the path; if so, we can also quick-error

Reply:
Like at the parser level?